### ISSUE TITLE
Improve the behavior of start/stop/halt when using a batch processor

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#2255]
Currently stopping a file reader that is processing a batch message will only finish the current message. We want to finish the entire file if the channel is stopped, and only stop processing immediately if the channel is halted.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-2316
Reporter: wayneh
Created: 2012-12-21T11:48:38.000-0800

**Comment by rbeckman-nextgen** [comment]
Note that currently the batch file will get moved to the destination directly even though it did not finish processing all of its messages.

Imported Comment. Original Details:
Author: wayneh
Created: 2012-12-21T11:52:49.000-0800

**Comment by rbeckman-nextgen** [comment]
Also, the channel is stopped and "Delete after read" is enabled, the file is lost even though it hasn't yet been fully read by the channel.

Imported Comment. Original Details:
Author: narupley
Created: 2013-09-05T16:14:00.000-0700

**Comment by rbeckman-nextgen** [comment]
Currently, if the channel is stopped while processing a batch message, any of the following can happen.

1. If a file is read but no messages are processed yet, the after processing action will be executed with no error message. This means a file can be deleted with no messages read.
2. If some messages have been processed, the channel will log an error and the after processing action will be executed.
3. If the second to last message has been processed, the channel will log an error and the ERROR action will be executed.

Imported Comment. Original Details:
Author: wayneh
Created: 2014-06-26T10:11:43.000-0700

**Comment by rbeckman-nextgen** [comment]
For connectors other than TCP, stopping the channel while processing a batch message will now finish all the messages in the batch before stopping. Halting will still stop the channel immediately and cut the batch short.

For TCP Receiver, stopping a channel still stops the batch immediately after the current message, even if there are still messages in the batch. 

Imported Comment. Original Details:
Author: wayneh
Created: 2014-07-24T17:44:30.000-0700

**Comment by rbeckman-nextgen** [comment]
Made several simple channels to test out the functionality. Using a file reader channel, I verified that I was able to stop the channel mid batch with the channel reading to completion then moving or deleting the channel. When the same channel is halted mid process, the channel stops reading and the file remains unchanged since it hasn't finished processing. Also tested this using other connector types and varying data types to ensure the message was read in its entirety.

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-08-29T11:23:33.000-0700
